Lib/test/support/__init__.py

```diff
@@ -2940,3 +2940,31 @@ def in_systemd_nspawn_sync_suppressed() -> bool:
         os.close(fd)
 
     return False
+
+def run_simple_async_fn(async_fn, /, *args, **kwargs):
+    coro = async_fn(*args, **kwargs)
+    try:
+        coro.send(None)
+    except StopIteration as e:
+        return e.value
+    else:
+        raise AssertionError("coroutine did not complete")
+    finally:
+        coro.close()
+
+
+@types.coroutine
+def async_yield(v):
+    return (yield v)
+
+
+def run_async_fn(async_fn, /, *args, **kwargs):
+    coro = async_fn(*args, **kwargs)
+    try:
+        while True:
+            try:
+                coro.send(None)
+            except StopIteration as e:
+                return e.value
+    finally:
+        coro.close()
```

Comment:
Could you please add these functions to `__all__`?
Though maybe it makes sense to give more appropriate names for `run_simple_async_fn/run_async_fn`, for me the difference between them is not obvious by reading their names

Reply:
claude suggests: `run_simple_async_fn -> run_async_once` and `run_async_fn -> run_async`

some more options:

# For run_simple_async_fn:
run_single_yield_async
execute_basic_async
run_atomic_async
run_one_step_async
execute_simple_async
run_async_once
run_bare_async
execute_async_minimal
run_async_single
run_sync_wrapped_async

# For run_async_fn:
run_async_complete
execute_async_full
run_async_chain
run_async_flow
execute_async
run_async_iter
run_async_through
execute_async_sequence
run_multi_yield_async
run_async


I like run_single_yield_async and run_multi_yield_async

Reply:
I like `run_single_yield_async` and `run_multi_yield_async` :)

Reply:
single_yield_async isn't correct, because it would raise a RuntimeError if there were even a single yield. It must be no_yield. I ran out of free credits though

Reply:
Okay, let's name it `run_no_yield_async`

Reply:
I prefer having the fn there because otherwise it sounds like "run thing asyncly" as opposed to "run category of async thing"

Reply:
I've pushed with some new names